src/sdk/pynni/nni/tuner.py

```diff
@@ -93,3 +93,16 @@ def _on_exit(self):
 
     def _on_error(self):
         pass
+
+    def extract_scalar_reward(self, value, scalar_key='default'):
+        if isinstance(value, float) or isinstance(value, int):
+            reward = value
+        elif isinstance(value, dict) and scalar_key in value:
+            reward = value[scalar_key]
+            if isinstance(reward, float) or isinstance(reward, int):
+                pass
```

Comment:
Suggest to check type more straight forward, such as:
```
if not isinstance(reward, (float, int)):
    raise...
```

Reply:
done